### PR TITLE
[SQL] Bumb sql sdk version, add output breaking change notice on SQL MI cmdlets and ignore API version in playback mode

### PR DESCRIPTION
**Pull request by milanbrkic-ms**
## Description
I have switched to latest SQL SDK where output model has changed and breaking change noticed was added to commands where that happened.

Also, BackupStorageRedundancy param now accept GeoZone as another valid value. 

A lot of test started failing because of sdk bump, so a lot of recorded files are part of this PR too.

## Checklist

- [X] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [X] The title of the PR is clear and informative
- [X] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [X] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by msJinLei** [comment]
@MilanBrkicFON Please resolve conflicts

**Comment by MilanBrkicFON** [comment]
@msJinLei done. Can you re-run pipeline and review change?

**Comment by MilanBrkicFON** [comment]
Hey @msJinLei what is going on with the pipeline ? Seems like jobs are stuck or even not started

**Comment by msJinLei** [comment]
> Hey @msJinLei what is going on with the pipeline ? Seems like jobs are stuck or even not started

I will check it soon

**Comment by msJinLei** [comment]
> Hey @msJinLei what is going on with the pipeline ? Seems like jobs are stuck or even not started

@MilanBrkicFON could you try to sync with main again and do a force push? 
https://github.com/Azure/azure-powershell/pull/16649

**Comment by msJinLei** [comment]
Please fix the following compilation error
```
    Az.ApplicationInsights -> /home/vsts/work/1/s/artifacts/Debug/Az.ApplicationInsights/ApplicationInsights.Autorest/bin/Az.ApplicationInsights.private.dll
    Az.Compute -> /home/vsts/work/1/s/artifacts/Debug/Az.Compute/Compute.Autorest/bin/Az.Compute.private.dll
    ScenarioTest.ResourceManager -> /home/vsts/work/1/s/tools/ScenarioTest.ResourceManager/bin/Debug/netcoreapp2.1/Microsoft.Azure.PowerShell.ScenarioTest.ResourceManager.dll
    TestFx -> /home/vsts/work/1/s/tools/TestFx/bin/Debug/netcoreapp2.1/Microsoft.Azure.PowerShell.TestFx.dll
    Network.Test -> /home/vsts/work/1/s/src/Network/Network.Test/bin/Debug/netcoreapp2.1/Microsoft.Azure.PowerShell.Cmdlets.Network.Test.dll
ScenarioTests/SqlTestsBase.cs(78,56): error CS1003: Syntax error, ',' expected [/home/vsts/work/1/s/src/Sql/Sql.Test/Sql.Test.csproj] [/home/vsts/work/1/s/build.proj]
  
  Build FAILED.
```

**Comment by msJinLei** [comment]
```
  "AssemblyFileName","ClassName","Target","Severity","ProblemId","Description","Remediation"
  "Az.Sql","Microsoft.Azure.Commands.Sql.TransparentDataEncryption.Cmdlet.AddAzureRmSqlManagedInstanceKeyVaultKey","Add-AzSqlInstanceKeyVaultKey","0","3010","The property 'BackupStorageRedundancy' of type 'Microsoft.Azure.Commands.Sql.ManagedInstance.Model.AzureSqlManagedInstanceModel' has been removed.","Add the property 'BackupStorageRedundancy' back to type 'Microsoft.Azure.Commands.Sql.ManagedInstance.Model.AzureSqlManagedInstanceModel'."
  "Az.Sql","Microsoft.Azure.Commands.Sql.VulnerabilityAssessmentSettings.Cmdlet.ClearSqlManagedInstanceVulnerabilityAssessmentSettings","Clear-AzSqlInstanceVulnerabilityAssessmentSetting","0","3010","The property 'BackupStorageRedundancy' of type 'Microsoft.Azure.Commands.Sql.ManagedInstance.Model.AzureSqlManagedInstanceModel' has been removed.","Add the property 'BackupStorageRedundancy' back to type 'Microsoft.Azure.Commands.Sql.ManagedInstance.Model.AzureSqlManagedInstanceModel'."
  "Az.Sql","Microsoft.Azure.Commands.Sql.InstanceActiveDirectoryOnlyAuthentication.Cmdlet.DisableAzureSqlInstanceActiveDirectoryOnlyAuthentication","Disable-AzSqlInstanceActiveDirectoryOnlyAuthentication","0","3010","The property 'BackupStorageRedundancy' of type 'Microsoft.Azure.Commands.Sql.ManagedInstance.Model.AzureSqlManagedInstanceModel' has been removed.","Add the property 'BackupStorageRedundancy' back to type 'Microsoft.Azure.Commands.Sql.ManagedInstance.Model.AzureSqlManagedInstanceModel'."
  "Az.Sql","Microsoft.Azure.Commands.Sql.AdvancedThreatProtection.Cmdlet.DisableAzureSqlManagedInstanceAdvancedDataSecurity","Disable-AzSqlInstanceAdvancedDataSecurity","0","3010","The property 'BackupStorageRedundancy' of type 'Microsoft.Azure.Commands.Sql.ManagedInstance.Model.AzureSqlManagedInstanceModel' has been removed.","Add the property 'BackupStorageRedundancy' back to type 'Microsoft.Azure.Commands.Sql.ManagedInstance.Model.AzureSqlManagedInstanceModel'."
  "Az.Sql","Microsoft.Azure.Commands.Sql.InstanceActiveDirectoryOnlyAuthentication.Cmdlet.EnableAzureSqlInstanceActiveDirectoryOnlyAuthentication","Enable-AzSqlInstanceActiveDirectoryOnlyAuthentication","0","3010","The property 'BackupStorageRedundancy' of type 'Microsoft.Azure.Commands.Sql.ManagedInstance.Model.AzureSqlManagedInstanceModel' has been removed.","Add the property 'BackupStorageRedundancy' back to type 'Microsoft.Azure.Commands.Sql.ManagedInstance.Model.AzureSqlManagedInstanceModel'."
  "Az.Sql","Microsoft.Azure.Commands.Sql.AdvancedThreatProtection.Cmdlet.EnableAzureSqlManagedInstanceAdvancedDataSecurity","Enable-AzSqlInstanceAdvancedDataSecurity","0","3010","The property 'BackupStorageRedundancy' of type 'Microsoft.Azure.Commands.Sql.ManagedInstance.Model.AzureSqlManagedInstanceModel' has been removed.","Add the property 'BackupStorageRedundancy' back to type 'Microsoft.Azure.Commands.Sql.ManagedInstance.Model.AzureSqlManagedInstanceModel'."
  "Az.Sql","Microsoft.Azure.Commands.Sql.TransparentDataEncryption.Cmdlet.GetAzureSqlDatabaseTransparentDataEncryptionActivity","Get-AzSqlDatabaseTransparentDataEncryptionActivity","0","1000","The cmdlet 'Get-AzSqlDatabaseTransparentDataEncryptionActivity' has been removed and no alias was found for the original cmdlet name.","Add the cmdlet 'Get-AzSqlDatabaseTransparentDataEncryptionActivity' back to the module, or add an alias to the original cmdlet name."
  "Az.Sql","Microsoft.Azure.Commands.Sql.ManagedInstance.Cmdlet.GetAzureSqlManagedInstance","Get-AzSqlInstance","0","3010","The property 'BackupStorageRedundancy' of type 'Microsoft.Azure.Commands.Sql.ManagedInstance.Model.AzureSqlManagedInstanceModel' has been removed.","Add the property 'BackupStorageRedundancy' back to type 'Microsoft.Azure.Commands.Sql.ManagedInstance.Model.AzureSqlManagedInstanceModel'."
  "Az.Sql","Microsoft.Azure.Commands.Sql.InstanceActiveDirectoryAdministrator.Cmdlet.GetAzureSqlInstanceActiveDirectoryAdministrator","Get-AzSqlInstanceActiveDirectoryAdministrator","0","3010","The property 'BackupStorageRedundancy' of type 'Microsoft.Azure.Commands.Sql.ManagedInstance.Model.AzureSqlManagedInstanceModel' has been removed.","Add the property 'BackupStorageRedundancy' back to type 'Microsoft.Azure.Commands.Sql.ManagedInstance.Model.AzureSqlManagedInstanceModel'."
  "Az.Sql","Microsoft.Azure.Commands.Sql.InstanceActiveDirectoryOnlyAuthentication.Cmdlet.GetAzureSqlInstanceActiveDirectoryOnlyAuthentication","Get-AzSqlInstanceActiveDirectoryOnlyAuthentication","0","3010","The property 'BackupStorageRedundancy' of type 'Microsoft.Azure.Commands.Sql.ManagedInstance.Model.AzureSqlManagedInstanceModel' has been removed.","Add the property 'BackupStorageRedundancy' back to type 'Microsoft.Azure.Commands.Sql.ManagedInstance.Model.AzureSqlManagedInstanceModel'."
  "Az.Sql","Microsoft.Azure.Commands.Sql.AdvancedThreatProtection.Cmdlet.GetAzureSqlManagedInstanceAdvancedDataSecurityPolicy","Get-AzSqlInstanceAdvancedDataSecurityPolicy","0","3010","The property 'BackupStorageRedundancy' of type 'Microsoft.Azure.Commands.Sql.ManagedInstance.Model.AzureSqlManagedInstanceModel' has been removed.","Add the property 'BackupStorageRedundancy' back to type 'Microsoft.Azure.Commands.Sql.ManagedInstance.Model.AzureSqlManagedInstanceModel'."
  "Az.Sql","Microsoft.Azure.Commands.Sql.ManagedDatabase.Cmdlet.GetAzureSqlManagedDatabase","Get-AzSqlInstanceDatabase","0","3010","The property 'BackupStorageRedundancy' of type 'Microsoft.Azure.Commands.Sql.ManagedInstance.Model.AzureSqlManagedInstanceModel' has been removed.","Add the property 'BackupStorageRedundancy' back to type 'Microsoft.Azure.Commands.Sql.ManagedInstance.Model.AzureSqlManagedInstanceModel'."
  "Az.Sql","Microsoft.Azure.Commands.Sql.TransparentDataEncryption.Cmdlet.GetAzureRmSqlManagedInstanceKeyVaultKey","Get-AzSqlInstanceKeyVaultKey","0","3010","The property 'BackupStorageRedundancy' of type 'Microsoft.Azure.Commands.Sql.ManagedInstance.Model.AzureSqlManagedInstanceModel' has been removed.","Add the property 'BackupStorageRedundancy' back to type 'Microsoft.Azure.Commands.Sql.ManagedInstance.Model.AzureSqlManagedInstanceModel'."
  "Az.Sql","Microsoft.Azure.Commands.Sql.TransparentDataEncryption.Cmdlet.GetAzureRmSqlManagedInstanceTransparentDataEncryptionProtector","Get-AzSqlInstanceTransparentDataEncryptionProtector","0","3010","The property 'BackupStorageRedundancy' of type 'Microsoft.Azure.Commands.Sql.ManagedInstance.Model.AzureSqlManagedInstanceModel' has been removed.","Add the property 'BackupStorageRedundancy' back to type 'Microsoft.Azure.Commands.Sql.ManagedInstance.Model.AzureSqlManagedInstanceModel'."
  "Az.Sql","Microsoft.Azure.Commands.Sql.VulnerabilityAssessmentSettings.Cmdlet.GetAzureSqlManagedInstanceVulnerabilityAssessmentSettings","Get-AzSqlInstanceVulnerabilityAssessmentSetting","0","3010","The property 'BackupStorageRedundancy' of type 'Microsoft.Azure.Commands.Sql.ManagedInstance.Model.AzureSqlManagedInstanceModel' has been removed.","Add the property 'BackupStorageRedundancy' back to type 'Microsoft.Azure.Commands.Sql.ManagedInstance.Model.AzureSqlManagedInstanceModel'."
  "Az.Sql","Microsoft.Azure.Commands.Sql.ManagedInstance.Cmdlet.NewAzureSqlManagedInstance","New-AzSqlInstance","0","3010","The property 'BackupStorageRedundancy' of type 'Microsoft.Azure.Commands.Sql.ManagedInstance.Model.AzureSqlManagedInstanceModel' has been removed.","Add the property 'BackupStorageRedundancy' back to type 'Microsoft.Azure.Commands.Sql.ManagedInstance.Model.AzureSqlManagedInstanceModel'."
  "Az.Sql","Microsoft.Azure.Commands.Sql.ManagedDatabase.Cmdlet.NewAzureSqlManagedDatabase","New-AzSqlInstanceDatabase","0","3010","The property 'BackupStorageRedundancy' of type 'Microsoft.Azure.Commands.Sql.ManagedInstance.Model.AzureSqlManagedInstanceModel' has been removed.","Add the property 'BackupStorageRedundancy' back to type 'Microsoft.Azure.Commands.Sql.ManagedInstance.Model.AzureSqlManagedInstanceModel'."
```

**Comment by MilanBrkicFON** [comment]
I needed one commit from release/2022-02-08 branch which is not in the main currently and that is why I merged that branch. When main branch is updated I will rebase

**Comment by MilanBrkicFON** [comment]
@msJinLei can you please review again ? The Breaking change notes were checked in with [previous release](https://github.com/Azure/azure-powershell/pull/16955), so I think I am good to deprecate commands now. What is the procedure ?

**Comment by BethanyZhou** [comment]
@MilanBrkicFON , please let me know if your PR is ready to review.

**Comment by BethanyZhou** [comment]
@MilanBrkicFON , please let me know if your PR is ready to review.

File: src/Sql/Sql/ManagedInstance/Cmdlet/GetAzureSqlManagedInstance.cs [hunk 1]
```diff
@@ -25,6 +27,11 @@ namespace Microsoft.Azure.Commands.Sql.ManagedInstance.Cmdlet
     /// <summary>
     /// Defines the Get-AzSqlInstance cmdlet
     /// </summary>
+    [CmdletOutputBreakingChange(
+        deprecatedCmdletOutputTypeName: typeof(AzureSqlManagedInstanceModel),
+        deprecateByVersion: "3.8.0",
+        DeprecatedOutputProperties = new String[] { "BackupStorageRedundancy" },
+        NewOutputProperties = new String[] { "CurrentBackupStorageRedundancy", "RequestedBackupStorageRedundancy" })]
```
[the remark this answers is not in the document]
Comment (BethanyZhou) [reply src/Sql/Sql/ManagedInstance/Cmdlet/GetAzureSqlManagedInstance.cs:34]: As we discussed in email. BackupStorageRedundancy is not going to be deprecated. Please keep the value of BackupStorageRedundancy as before.

[the remark this answers is not in the document]
Comment (BethanyZhou) [reply src/Sql/Sql/ManagedInstance/Cmdlet/GetAzureSqlManagedInstance.cs:34]: As we discussed in email. BackupStorageRedundancy is not going to be deprecated. Please keep the value of BackupStorageRedundancy as before.

**Comment by MilanBrkicFON** [comment]
@BethanyZhou it is ready

File: src/Sql/Sql/ManagedInstance/Cmdlet/GetAzureSqlManagedInstance.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (MilanBrkicFON) [reply src/Sql/Sql/ManagedInstance/Cmdlet/GetAzureSqlManagedInstance.cs:34]: left logic for assigning all 3 properties and left breaking change notes

File: src/Sql/Sql/ChangeLog.md [hunk 2]
```diff
@@ -17,7 +17,11 @@
     * Overview of change #1
         - Additional information about change #1
 -->
+
 ## Upcoming Release
+* Output braking change for Managed Instance CRUD commands
+    - Instead of BackupStorageRedundancy, we now return CurrentBackupStorageRedundancy and RequestedBackupStorageRedundancy.
+    - `Get-AzSqlDatabaseTransparentDataEncryptionActivity` is now removed.
```
[the remark this answers is not in the document]
Comment (MilanBrkicFON) [reply src/Sql/Sql/ChangeLog.md]: Done

File: src/Sql/Sql/ChangeLog.md [hunk 3]
```diff
@@ -17,7 +17,10 @@
     * Overview of change #1
         - Additional information about change #1
 -->
+
```
Comment (BethanyZhou) [inline src/Sql/Sql/ChangeLog.md:20]:
```suggestion
```

File: src/Sql/Sql.Test/ScenarioTests/DatabaseBackupTests.cs [hunk 4]
```diff
@@ -193,14 +193,14 @@ public void TestShortTermRetentionPolicy()
             }
         }
 
-        [Fact]
+        [Fact(Skip = "Location 'East US 2 EUAP' is not accepting creation of new Windows Azure SQL Database servers at this time.'")]
```
Comment (BethanyZhou) [inline src/Sql/Sql.Test/ScenarioTests/DatabaseBackupTests.cs:196]: Could you please try another location to run this test?

File: src/Sql/Sql.Test/ScenarioTests/RecommendedActionTests.cs [hunk 5]
```diff
@@ -27,21 +27,21 @@ public RecommendedActionTests(ITestOutputHelper output) : base(output)
 
         #region Server Recommended Action Tests
 
-        [Fact]
+        [Fact(Skip = "Resource group 'WIRunnersProd' could not be found.")]
```
Comment (BethanyZhou) [inline src/Sql/Sql.Test/ScenarioTests/RecommendedActionTests.cs:30]: Could you please re-create the resource group and the resources needed to run this test?

File: src/Sql/Sql.Test/ScenarioTests/TransparentDataEncryptionCrudTests.cs [hunk 6]
```diff
@@ -28,7 +28,7 @@ public TransparentDataEncryptionCrudTests(ITestOutputHelper output) : base(outpu
             };
         }
 
-        [Fact]
+        [Fact(Skip = "Object reference not set to an instance of an object.")]
```
Comment (BethanyZhou) [inline src/Sql/Sql.Test/ScenarioTests/TransparentDataEncryptionCrudTests.cs:31]: Could you please check the root cause why this test was failed? All tests aim to ensure the correctness and quality of codes. We need to make sure tests can be passed if no special reason.

File: src/Sql/Sql/ManagedInstance/Cmdlet/NewAzureSqlManagedInstance.cs [hunk 7]
```diff
@@ -548,7 +554,7 @@ protected override IEnumerable<Model.AzureSqlManagedInstanceModel> ApplyUserInpu
                 DnsZonePartner = this.DnsZonePartner,
                 InstancePoolName = this.InstancePoolName,
                 MinimalTlsVersion = this.MinimalTlsVersion,
-                BackupStorageRedundancy = this.BackupStorageRedundancy,
```
Comment (BethanyZhou) [inline src/Sql/Sql/ManagedInstance/Cmdlet/NewAzureSqlManagedInstance.cs:551]: Current change will cause the value of BackupStorageRedundancy is null. It's unconsumable value for customer. Can we keep the value of BackupStorageRedundancy same as before?

File: src/Sql/Sql.Test/ScenarioTests/DatabaseBackupTests.cs (hunk 4 above)
Comment (MilanBrkicFON) [reply src/Sql/Sql.Test/ScenarioTests/DatabaseBackupTests.cs:196]: I have tried west central us which should be close to euap regions and provisioning is not supported. 
`"Provisioning of database with zone backup redundancy is not supported in this location."`

File: src/Sql/Sql/ManagedInstance/Cmdlet/NewAzureSqlManagedInstance.cs (hunk 7 above)
Comment (MilanBrkicFON) [reply src/Sql/Sql/ManagedInstance/Cmdlet/NewAzureSqlManagedInstance.cs:551]: This part of the code is sending request, and new API version is not supporting `BackupStorageRedundancy` as requested parameter, but `RequestedBackupStorageRedundancy`. Regarding output parameters they are composed in Adatpter file and I left all 3 properties in it - https://github.com/Azure/azure-powershell/pull/16611/files/2fd87022420f3e3d4439b3951e14bf6e5815dd09#diff-64f21e81171c7e74c8f190ed2acda574fe67cd21e235b27c629be142ccc6c393R255

**Comment by MilanBrkicFON** [comment]
Hi @BethanyZhou , regarding skipped tests:

- DatabaseBackupTests - seems like no other region is supporting this feature
- RecommendedActionTests - I am working with the owners, but this could take a while to setup. What is your take?
- TransparentDataEncryptionCrudTests - they definitely have bugs in sdk, because I see other contributors were skipping other tests because of the same problem. What happens is that response is empty and that is because API returns 202 which sdk does not know how to parse. we have 2 options:

 A. Skip the test
 B. I can fix the break by adding condition to code, but then command will show inaccurate data

File: src/Sql/Sql.Test/ScenarioTests/RecommendedActionTests.cs (hunk 5 above)
Comment (MilanBrkicFON) [reply src/Sql/Sql.Test/ScenarioTests/RecommendedActionTests.cs]: Done

File: src/Sql/Sql/ChangeLog.md [hunk 8]
```diff
@@ -18,6 +18,8 @@
         - Additional information about change #1
 -->
 ## Upcoming Release
+* [Breaking change] Removed `Get-AzSqlDatabaseTransparentDataEncryptionActivity`
+* Added property `CurrentBackupStorageRedundancy` and `RequestedBackupStorageRedundancy` in  the outputs of Managed Instance CRUD commands
```
Comment (BethanyZhou) [inline src/Sql/Sql/ChangeLog.md:22]:
```suggestion
* Added property `CurrentBackupStorageRedundancy` and `RequestedBackupStorageRedundancy` in the outputs of Managed Instance CRUD commands
```

File: src/Sql/Sql/ManagedInstance/Cmdlet/NewAzureSqlManagedInstance.cs [hunk 9]
```diff
@@ -29,12 +29,18 @@
...
+using Microsoft.WindowsAzure.Commands.Common.CustomAttributes;
 
 namespace Microsoft.Azure.Commands.Sql.ManagedInstance.Cmdlet
 {
     /// <summary>
     /// Defines the New-AzSqlInstance cmdlet
     /// </summary>
+    [CmdletOutputBreakingChange(
+        deprecatedCmdletOutputTypeName: typeof(AzureSqlManagedInstanceModel),
+        deprecateByVersion: "3.8.0",
```
Comment (BethanyZhou) [inline src/Sql/Sql/ManagedInstance/Cmdlet/NewAzureSqlManagedInstance.cs:41]:
```suggestion
        deprecateByVersion: "4.0.0",
```

File: src/Sql/Sql.Test/ScenarioTests/TransparentDataEncryptionCrudTests.cs (hunk 6 above)
Comment (MilanBrkicFON) [reply src/Sql/Sql.Test/ScenarioTests/TransparentDataEncryptionCrudTests.cs]: Seems like this command is a bit regressed as now API returns 202 code which sdk doesn't know to handle and returns empty body. I have fixed command, to handle empty body, but return information will not be correct until sdk is fixed.

**Review by BethanyZhou — APPROVED** [review]
LGTM

**Pull request merged**